### PR TITLE
Aavechan/ghol mextension

### PR DESCRIPTION
**Pull request by marczeller**

### CODE REVIEW

File: tests/EmissionExtensionTestARBLM.t.sol [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+        reward: REWARD_ASSET,
+        transferStrategy: TRANSFER_STRATEGY,
+        rewardOracle: REWARD_ORACLE
+      });
+    }
+
+    return configs;
+  }
+
+  function _getEmissionsPerAsset() internal pure returns (EmissionPerAsset[] memory) {
```
Comment (brotherlymite) [inline tests/EmissionExtensionTestARBLM.t.sol:202]: same here, should be removed

File: tests/EmissionExtensionTestARBLM.t.sol [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+    newDistributionEndPerAsset.asset = GHO_A_TOKEN;
+    newDistributionEndPerAsset.reward = REWARD_ASSET;
+    newDistributionEndPerAsset.newDistributionEnd = _toUint32(
+      block.timestamp + NEW_DURATION_DISTRIBUTION_END
+    );
+
+    return newDistributionEndPerAsset;
+  }
+
+  function _toUint32(uint256 value) internal pure returns (uint32) {
```
Comment (brotherlymite) [inline tests/EmissionExtensionTestARBLM.t.sol:144]: unused, should be removed

File: tests/EmissionExtensionTestARBLM.t.sol [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+  address constant EMISSION_ADMIN = 0xac140648435d03f784879cd789130F22Ef588Fcd; // ACI
+  address constant REWARD_ASSET = ARB;
+
+  uint256 constant NEW_TOTAL_DISTRIBUTION = 72_800 ether;
+  uint88 constant NEW_DURATION_DISTRIBUTION_END = 15 days;
+
+  IEACAggregatorProxy constant REWARD_ORACLE = IEACAggregatorProxy(ARB_ORACLE);
+
+  ITransferStrategyBase constant TRANSFER_STRATEGY =
+    ITransferStrategyBase(0x991bf7661F1F2695ac8AEFc4F9a19718d6424dc0); // new deployed strategy
```
Comment (brotherlymite) [inline tests/EmissionExtensionTestARBLM.t.sol:50]: unused, should be removed

File: tests/EmissionExtensionTestARBLM.t.sol [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+  //  emission Total emission of a `reward` token during the whole distribution duration defined
+  // E.g. With an emission of 10_000 MATICX tokens during 1 month, an emission of 50% for variableDebtPolWMATIC would be
+  // 10_000 * 1e18 * 50% / 30 days in seconds = 1_000 * 1e18 / 2_592_000 = ~ 0.0003858 * 1e18 MATICX per second
+
+  address constant GHO_A_TOKEN = AaveV3ArbitrumAssets.GHO_A_TOKEN;// TODO: hardcoded for now will use lib when address book is updated
+  address constant ARB_ORACLE = AaveV3ArbitrumAssets.ARB_ORACLE;
+  address constant ARB = AaveV3ArbitrumAssets.ARB_UNDERLYING;
+
+
+  struct EmissionPerAsset {
```
Comment (brotherlymite) [inline tests/EmissionExtensionTestARBLM.t.sol:24]: unused

File: tests/EmissionExtensionTestARBLM.t.sol (hunk 2 above)
Comment (marczeller) [reply tests/EmissionExtensionTestARBLM.t.sol:144]: L137

Comment (brotherlymite) [reply tests/EmissionExtensionTestARBLM.t.sol:144]: my bad

File: tests/EmissionExtensionTestARBLM.t.sol [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+    address asset;
+    uint256 emission;
+  }
```
Comment (brotherlymite) [inline tests/EmissionExtensionTestARBLM.t.sol:27]:
```suggestion
```

File: tests/EmissionExtensionTestARBLM.t.sol [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+    address[] memory assets = new address[](1);
+    assets[0] = asset;
+
+    IAaveIncentivesController(AaveV3Arbitrum.DEFAULT_INCENTIVES_CONTROLLER).claimRewards(
+      assets,
+      type(uint256).max,
+      whale,
+      REWARD_ASSET
+    );
+
```
Comment (brotherlymite) [inline tests/EmissionExtensionTestARBLM.t.sol:164]:
```suggestion
```

File: tests/EmissionExtensionTestARBLM.t.sol [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+    uint256 deviationAccepted = expectedReward; // Approx estimated rewards with current emission in 1 month
+    assertApproxEqAbs(
+      balanceBefore,
+      balanceAfter,
+      deviationAccepted,
+      'Invalid delta on claimed rewards'
+    );
+
+    vm.stopPrank();
+  }
```
Comment (brotherlymite) [inline tests/EmissionExtensionTestARBLM.t.sol:165]:
```suggestion
```

File: tests/EmissionExtensionTestARBLM.t.sol [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+    return newDistributionEndPerAsset;
+  }
+
+  function _toUint32(uint256 value) internal pure returns (uint32) {
+    require(value <= type(uint32).max, "SafeCast: value doesn't fit in 32 bits");
+    return uint32(value);
+  }
+
+
+
```
Comment (brotherlymite) [inline tests/EmissionExtensionTestARBLM.t.sol:144]:
```suggestion

```

**Pull request closed without merging**